auth/src/main/java/com/firebase/ui/auth/util/signincontainer/SmartLock.java

```diff
@@ -17,11 +26,64 @@
 import java.util.Collections;
 import java.util.List;
 
-/**
- * Helper class to deal with Smartlock Flows.
- */
-public class SmartLockUtil {
-    private static final String TAG = "SmartLockUtil";
+public abstract class SmartLock<R extends Result> extends BaseFragment implements
```

Comment:
nit: I don't think "SmartLock" is the right name for this class.  This class is mostly just a Fragment class that cleans up its own dialogs and GoogleApiClients, there is no SmartLock/CredentialsAPI functionality here.